File: nodejs/openai-prompt/handler.js
import axios from 'axios';

// Handler function
async function handler(req, res) {
  try {
    let { prompt, max_tokens, temperature, model, token } = req.body;
    token = process.env.OPENAI_API_KEY || token;

    // Validate input
    if (!prompt || typeof prompt !== "string") {
      return res.status(400).json({ error: "Invalid prompt" });
    }

    // Validate token
    if (!token) {
      return res.status(400).json({ error: "OpenAI API key is required" });
    }

    // Prepare request options
    const options = {
      method: 'POST',
      url: 'https://api.openai.com/v1/chat/completions',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${token}`, // Replace with your OpenAI API key
      },
      data: {
        model: model || 'gpt-3.5-turbo',
        messages: [{ role: 'user', content: prompt }],
        max_tokens: max_tokens || 150,
        temperature: temperature || 0.7,
      },
    };

    // Make HTTP request
    const response = await axios(options);

    // Return response from OpenAI
    res.json(response.data);
  } catch (error) {
    console.error("Error:", error);
    res.status(500).json({ error: "Internal Server Error" });
  }
}

export { handler };

File: python/openai-prompt/handler.py
import json
import requests

# Function to handle requests
def handler(request):
    try:
        # Get JSON data from the request
        request_body = request.get_json(force=True, silent=True)
        
        # Extract required parameters from the request body
        prompt = request_body.get('prompt')
        max_tokens = request_body.get('max_tokens', 150)
        temperature = request_body.get('temperature', 0.7)
        model = request_body.get('model', 'gpt-3.5-turbo')
        token = request_body.get('token')

        # Validate input
        if not prompt or not isinstance(prompt, str):
            return {'error': 'Invalid prompt'}

        # Prepare request options
        url = 'https://api.openai.com/v1/chat/completions'
        headers = {
            'Content-Type': 'application/json',
            'Authorization': f'Bearer {token}'
        }

        # Make HTTP request
        response_body = make_http_request(url, headers, json.dumps({
            'model': model,
            'messages': [{'role': 'user', 'content': prompt}],
            'max_tokens': max_tokens,
            'temperature': temperature
        }))

        # Parse the response body
        res_json = json.loads(response_body['body'])
        return res_json
    except Exception as e:
        # Handle exceptions
        return {'error': 'Internal Server Error'}

# Function to make HTTP request
def make_http_request(url, headers, body):
    response = requests.post(url, headers=headers, data=body)
    return {
        'status': response.status_code,
        'headers': response.headers,
        'body': response.text
    }
